theano/tensor/nnet/tests/test_sigm.py

```diff
@@ -347,14 +334,14 @@ def test_local_hard_sigmoid(self):
 
         mode = self.get_mode('local_hard_sigmoid')
         f = theano.function([x], s, mode=mode)
-        assert hasattr(f.maker.fgraph.outputs[0].tag, 'trace')
+        assert check_stack_trace(f, ops_to_check=sigmoid)
         topo = f.maker.fgraph.toposort()
         assert topo[0].op == sigmoid
         assert len(topo) == 1
 
         mode = self.get_mode().including('local_hard_sigmoid')
         f = theano.function([x], s, mode=mode)
-        assert hasattr(f.maker.fgraph.outputs[0].tag, 'trace')
+        assert check_stack_trace(f, ops_to_check='all')
```

Comment:
The issue here is that the "hard_sigmoid" replacement gets fused with other elemwise operations, and elemwise fusion does not preserve the stack yet.
The way to solve that here would be to exclude "fusion".
Also, `clip` is going to be made inplace by an opt that does not preserve the stack trace yet.

In the end, I think the following should work:
```python
# For checking that local_hard_sigmoid preserves the stack trace
mode2 = mode.excluding('fusion').excluding('inplace')
f2 = theano.function([x], s, mode=mode2)
self.assertTrue(check_stack_trace(f2, ops_to_check=theano.tensor.clip)
```